File: data/derived_cot/rq3_prompting/results/translation/r1/java_to_py/code_output/Warehouse.py
class Warehouse:
    class Product:
        def __init__(self, name: str, quantity: int):
            self.name = name
            self.quantity = quantity

        def addQuantity(self, quantity: int):
            self.quantity += quantity

        def getQuantity(self) -> int:
            return self.quantity

    class Order:
        def __init__(self, productId: int, quantity: int, status: str):
            self.productId = productId
            self.quantity = quantity
            self.status = status

        def getStatus(self) -> str:
            return self.status

        def setStatus(self, status: str):
            self.status = status

    def __init__(self):
        self.inventory = {}
        self.orders = {}

    def addProduct(self, productId: int, name: str, quantity: int):
        if productId in self.inventory:
            self.inventory[productId].addQuantity(quantity)
        else:
            self.inventory[productId] = self.Product(name, quantity)

    def updateProductQuantity(self, productId: int, quantity: int):
        if productId in self.inventory:
            self.inventory[productId].addQuantity(quantity)

    def getProductQuantity(self, productId: int) -> int:
        if productId in self.inventory:
            return self.inventory[productId].getQuantity()
        else:
            return -1

    def createOrder(self, orderId: int, productId: int, quantity: int) -> bool:
        if productId in self.inventory and self.inventory[productId].getQuantity() >= quantity:
            self.inventory[productId].addQuantity(-quantity)
            self.orders[orderId] = self.Order(productId, quantity, "Shipped")
            return True
        else:
            return False

    def changeOrderStatus(self, orderId: int, status: str) -> bool:
        if orderId in self.orders:
            self.orders[orderId].setStatus(status)
            return True
        else:
            return False

    def trackOrder(self, orderId: int) -> str:
        if orderId in self.orders:
            return self.orders[orderId].getStatus()
        else:
            return None

File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/java_to_cpp/code_output/Warehouse.cpp
#include <unordered_map>
#include <string>
#include <optional>

class Warehouse {
private:
    struct Product {
        std::string name;
        int quantity;

        Product(std::string name, int quantity) : name(std::move(name)), quantity(quantity) {}

        void addQuantity(int qty) {
            this->quantity += qty;
        }

        int getQuantity() const {
            return quantity;
        }
    };

    struct Order {
        int productId;
        int quantity;
        std::string status;

        Order(int productId, int quantity, std::string status) : productId(productId), quantity(quantity), status(std::move(status)) {}

        std::string getStatus() const {
            return status;
        }

        void setStatus(std::string stat) {
            this->status = std::move(stat);
        }
    };

    std::unordered_map<int, Product> inventory;
    std::unordered_map<int, Order> orders;

public:
    Warehouse() = default;

    void addProduct(int productId, const std::string& name, int quantity) {
        auto it = inventory.find(productId);
        if (it != inventory.end()) {
            it->second.addQuantity(quantity);
        } else {
            inventory.emplace(productId, Product(name, quantity));
        }
    }

    void updateProductQuantity(int productId, int quantity) {
        auto it = inventory.find(productId);
        if (it != inventory.end()) {
            it->second.addQuantity(quantity);
        }
    }

    int getProductQuantity(int productId) {
        auto it = inventory.find(productId);
        if (it != inventory.end()) {
            return it->second.getQuantity();
        } else {
            return -1;
        }
    }

    bool createOrder(int orderId, int productId, int quantity) {
        auto it = inventory.find(productId);
        if (it != inventory.end() && it->second.getQuantity() >= quantity) {
            it->second.addQuantity(-quantity);
            orders.emplace(orderId, Order(productId, quantity, "Shipped"));
            return true;
        } else {
            return false;
        }
    }

    bool changeOrderStatus(int orderId, const std::string& status) {
        auto it = orders.find(orderId);
        if (it != orders.end()) {
            it->second.setStatus(status);
            return true;
        } else {
            return false;
        }
    }

    std::optional<std::string> trackOrder(int orderId) {
        auto it = orders.find(orderId);
        if (it != orders.end()) {
            return it->second.getStatus();
        } else {
            return std::nullopt;
        }
    }
};